front/app/api/notifications/types.ts

```diff
@@ -180,12 +182,42 @@ export interface IMentionInCommentNotificationData
     initiating_user_first_name: string | null;
     initiating_user_last_name: string | null;
     initiating_user_slug: string | null;
-    post_type: 'Initiative' | 'Idea';
+    post_type: PostType;
     post_slug: string | null;
     post_title_multiloc: Multiloc;
   };
 }
 
+interface IInternalCommentBaseAttributes {
+  read_at: string | null;
+  created_at: string;
+  initiating_user_first_name: string | null;
+  initiating_user_last_name: string | null;
+  initiating_user_slug: string | null;
+  post_type: PostType;
+  post_slug: string | null;
+  post_title_multiloc: Multiloc;
+  post_id: string;
+}
+
+type InternalCommentType =
+  | 'mention_in_internal_comment'
+  | 'internal_comment_on_your_internal_comment'
+  | 'internal_comment_on_idea_assigned_to_you'
+  | 'internal_comment_on_initiative_assigned_to_you'
+  | 'internal_comment_on_idea_you_moderate'
+  | 'internal_comment_on_idea_you_commented_internally_on'
+  | 'internal_comment_on_initiative_you_commented_internally_on'
+  | 'internal_comment_on_unassigned_unmoderated_idea'
+  | 'internal_comment_on_unassigned_initiative';
+
+export interface IInternalCommentNotificationData
+  extends IBaseNotificationData {
+  attributes: {
+    type: InternalCommentType;
+  } & IInternalCommentBaseAttributes;
+}
```

Comment:
```suggestion
export interface IInternalCommentNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: InternalCommentType;
    read_at: string | null;
    created_at: string;
    initiating_user_first_name: string | null;
    initiating_user_last_name: string | null;
    initiating_user_slug: string | null;
    post_type: PostType;
    post_slug: string | null;
    post_title_multiloc: Multiloc;
    post_id: string;
  };
}
```

Shall we put the attributes here directly as `type` is also part of the attributes? Also easier to read if you don't have to move up.